### PR TITLE
usm: Refactor nodejs monitor to use new uprobe attacher

### PR DESCRIPTION
**Pull request by gjulianm**
### What does this PR do?

This PR applies the new uprobe attacher introduced in https://github.com/DataDog/datadog-agent/pull/27663 to the NodeJS monitor.

### Describe how to test/QA your changes

Validated in system-probe load testing environment, using the soak_test and processes scenarios. Base version is agent release 7.57

<details>
<summary>soak_test</summary>

Summary: similar accuracy results, no significant difference in resource usage.

[Dashboard](https://dddev.datadoghq.com/dashboard/gqu-2wz-asy/usm-performance-evaluation-20?fromUser=false&refresh_mode=paused&tpl_var_base_agent-env%5B0%5D=gj-upa-node-soak-base&tpl_var_client-service%5B0%5D=node-k6-client-http-tls&tpl_var_compare_agent-env%5B0%5D=gj-upa-node-soak-new&tpl_var_http.version%5B0%5D=http%2F1.1&tpl_var_kube_cluster_name%5B0%5D=sp-test-env-1&tpl_var_server-service%5B0%5D=node-httpbin-tls&tpl_var_tls.library%5B0%5D=nodejs&from_ts=1726510874490&to_ts=1726512770000&live=false)

Resource usage:
Screenshot 2024-09-16 at 14 53 26

Accuracy for base:
Screenshot 2024-09-16 at 14 54 58

Accuracy for new  (this branch):
Screenshot 2024-09-16 at 14 54 50

Accuracy comparison:

Screenshot 2024-09-16 at 14 54 35

</details>

<details><summary>processes</summary>

Summary: negligible resource usage differences.

[Dashboard](https://dddev.datadoghq.com/dashboard/gqu-2wz-asy/usm-performance-evaluation-20?fromUser=true&refresh_mode=paused&tpl_var_base_agent-env%5B0%5D=gj-upa-node-proc-base&tpl_var_client-service%5B0%5D=node-k6-client-http-tls&tpl_var_compare_agent-env%5B0%5D=gj-upa-node-proc-new&tpl_var_http.version%5B0%5D=http%2F1.1&tpl_var_kube_cluster_name%5B0%5D=sp-test-env-1&tpl_var_server-service%5B0%5D=node-httpbin-tls&tpl_var_tls.library%5B0%5D=nodejs&from_ts=1726514056000&to_ts=1726514929820&live=false)

Resource usage:

Screenshot 2024-09-16 at 16 08 59

</details>

### CODE REVIEW

File: pkg/network/usm/nodejs.go [hunk 1]
```diff
@@ -111,35 +106,33 @@ var (
...
+		}},
+		EbpfConfig:                     &c.Config,
+		ExcludeTargets:                 uprobes.ExcludeSelf | uprobes.ExcludeInternal | uprobes.ExcludeBuildkit | uprobes.ExcludeContainerdTmp,
+		EnablePeriodicScanNewProcesses: true,
+	}
 
-		// Callbacks
-		registerCB:   addHooks(mgr, procRoot, nodeJSProbes),
-		unregisterCB: removeHooks(mgr, nodeJSProbes),
+	attacher, err := uprobes.NewUprobeAttacher(NodeJsAttacherName, attachCfg, mgr, nil, &uprobes.NativeBinaryInspector{})
```
Comment (guyarb) [inline pkg/network/usm/nodejs.go:129]: nit: providing `nil` (4th parameter) is unclear, maybe create a definition nopOnAttachCB (which equals to nil) to ease reading

File: pkg/network/usm/nodejs.go [hunk 2]
```diff
@@ -111,35 +106,33 @@ var (
...
+	if err != nil {
+		log.Errorf("Cannot create uprobe attacher: %v", err)
+	}
```
Comment (guyarb) [inline pkg/network/usm/nodejs.go:132]: if we failed creating the attacher, why do we continue?

Comment (gjulianm) [reply pkg/network/usm/nodejs.go]: An overlook. I changed the code to check and return the error if it happens

**Review by val06 — APPROVED** [review]

File: pkg/network/usm/ebpf_ssl.go [hunk 3]
```diff
@@ -461,11 +461,16 @@ func newSSLProgramProtocolFactory(m *manager.Manager) protocols.ProtocolFactory
 			}
 		}
 
+		nodejs, err := newNodeJSMonitor(c, m)
+		if err != nil {
+			return nil, fmt.Errorf("error initializing nodejs monitor: %s", err)
```
Comment (guyarb) [inline pkg/network/usm/ebpf_ssl.go:466]:
```suggestion
			return nil, fmt.Errorf("error initializing nodejs monitor: %w", err)
```

**Review by guyarb — APPROVED** [review]

**Pull request merged**